### DISCUSSION TITLE
Render all items with a certain tag through one menu item

### DISCUSSION THREAD

**Discussion by LordRazen** [joomengine/Joomla-Component-Builder#743] [Q&A] [answered]
Hello, 

I coded a component with an Admin View which includes a tag field so the items can be tagged. 

Now I wonder how to create a generic "tag" page, where only the items with this certain tag are shown. 

I could get it to work if I create a static menu item and select the tag as a menu parameter, but since I dont want to create a menu item for every tag, I would like to do this the generic way: Whenever you click on a tag in an item view, you go to a certain menuitem like this:

www.website.com/component/tag/nameofthetag

So the "tag" menu item is like a general menu for all tags to be rendered. 

Kinda lost here, a rough concept would do it ^^ Thanks a lot already.

**Comment by Llewellynvdm** [comment, marked as answer]
I know this may not be that helpful, but we are using tags in the [Sermon Distributor](https://github.com/Llewellynvdm/Joomla-Sermon-Distributor) component. You can pull that JCB package and see how it was done. That is the direction I took... but the options are endless. Joomla has its own TAG front-end, I have not made use of it in a 3rd party extension as yet. But I know that it exist and it useful... but not sure what all the conventions are, and so can not be of more help at this time.

**Reply by LordRazen** [reply]
Ok still thanks, will have a look in the Component ^^

**Comment by dormasta** [comment]
Hi, 
I choose the way through the Dynamic Get:
Name/custom Name: getTags
getType: getCustoms
**Main tab:**
Main Source: Joomla DB
DB Table (main): xxxx_tags
Selection: a.id AS tag_id       a.title AS tag_title

**Joint tab:**
Join DB Tables:

**first):**
DB Table: xxxx_contentitem_tag_map
Return Row Type: Single
AS: b
Type: join INNER
On Field: a.id
Comparaison Operators: equal
Join Field: b.tag_id
Selection: b.type_alias AS contentitem_alias

**second):**
DB Table: xxxx_your_object
Return Row Type: Single
AS: c
Type: join INNER
On Field: a.id
Comparaison Operators: equal
Join Field: b.content_item_id
Selection: c.id AS your_object_id


which produces something like:
```
SELECT c.id, title FROM xxxx_tags a
    INNER JOIN xxxx_contentitem_tag_map b
        on a.id = b.tag_id
    INNER JOIN xxxx_your_object c
        on  c.id = b.content_item_id;
```
